snuba/utils/streams/synchronized.py

```diff
@@ -86,8 +90,18 @@ def __init__(
         ] = Synchronized({group: {} for group in commit_log_groups})
 
         self.__commit_log_worker_stop_requested = Event()
+        self.__commit_log_worker_subscription_received = Event()
         self.__commit_log_worker = execute(self.__run_commit_log_worker)
 
+        logger.debug("Waiting for commit log consumer to receieve assignment...")
+        while not self.__commit_log_worker_subscription_received.wait(0.1):
```

Comment:
> This changes is only needed so that the DummyConsumer can still work in test_synchronnized after the change to DummyConsumer itself.

Yep.

> This is already behaving correctly today in prod. Is that correct?

Yes (as long as you mean "correctly" as in the invariants of the synchronized consumer are maintained, not specifically this behavior where the synchronized consumer waits for the assignment to occur before initializing.)

> And that's because the on_assign method is now only called (when using the DummyConsumer) when `self.__commit_log_consumer.poll(0.1)` is executed. Is this right?

Yes. I suppose another option here would be for the consumer test to wait until an assignment is received (potentially after multiple calls to `poll` rather than assuming a subscription will occur after the first `poll` call.